### ISSUE TITLE
Color picker enchantments

### ISSUE THREAD

**Issue by Viza74** [nurpax/petmate#49]
You have to move the mouse so far away to pick a color from the palette... Some suggestions to remedy that:

- shortcut keys to change color (like wasd for chars now): q-e keys to select next/previous color in the palette

- a different approach (and more work :) ) : a popup palette under the cursor. While holding "c" a 4x4 grid of colors pops up under the cursor, the cursor is in the center. Releasing c, or clicking a color hides the popup.

- ability to change palette sorting: right now the colors in the palette are in the order as the original color codes (and thats fine). Allow to switch to a palette sorted by brightness of the colors. Or maybe even custom gradients?... Thats even more work :)
There is a good reference image in this article: http://www.aaronbell.com/secret-colours-of-the-commodore-64/

**Comment by nurpax** [comment]
I also noticed that the border/bg color selector is kind of not great. These would work better with a 4x4 grid. 

I guess you talked about char color selector which is on the right side of the screen?

**Comment by Viza74** [comment]
Yes, the functionality of the color palette under the character... uhm.. palette.

**Comment by Viza74** [comment]
ohh: my issue with the border/bg color picker is not a usability one, because you change those setting relativey infrequelty (like once for a screen?...)
The problem is that it got too much emphasis in the current UI for its function/importance (prime "real estate" in the left toolbar along with all the frequenlty used tools).
Maybe under the screen thumbnails, with smaller icons? That would strengten the idea of a "screen-wide" setting.
Or simply just over the character palette. Because... there is enough not used space there. :)

**Comment by nurpax** [comment]
Yep the awesome @Esshahn already also thought about moving border/bg pickers to the top part of the UI in #43.

**Comment by nurpax** [comment]
Hmm, I implemented current color cycling with meta-left, meta-right (not q, e).  Maybe q,e would be better since it's next to asdw, like @Viza74 suggests?  Did the work as part of #58.

**Comment by Viza74** [comment]
Yepp, cmd-left/right is pretty uncomfortable especially on my external keyboard. It is bearably on the built macbook one, but on the ext. keyboard the cmd and arrow keys are quite far away from each other (and from wasd).

**Comment by ftoledo** [comment]
please add shortcut to pick current character color under cursor (on text draw mode)
